lib/ansible/module_utils/basic.py

```diff
@@ -1338,6 +1351,23 @@ def _check_argument_values(self):
                 continue
             if isinstance(choices, SEQUENCETYPE):
                 if k in self.params:
+                    # PyYaml converts certain strings to bools.  If we can unambiguously convert back, do so before checking the value.  If we can't figure this out, module author is responsible.
+                    lowered_choices = None
+                    if self.params[k] == 'False':
```

Comment:
shouldn't you only check if it does not already match the choices? Just in case 'False' or 'True' are valid choices